src/lib.rs

```diff
@@ -21,18 +21,12 @@ pub fn plugin_registrar(reg: &mut Registry) {
     reg.register_macro("speculate", expand_speculate);
 }
 
-fn expand_speculate(cx: &mut ExtCtxt, _sp: Span, tokens: &[TokenTree]) -> Box<MacResult + 'static> {
+fn expand_speculate(cx: &mut ExtCtxt, sp: Span, tokens: &[TokenTree]) -> Box<MacResult + 'static> {
+    let mod_name = format!("speculate #{}", sp.lo().0);
+    
     let mut parser = stream_to_parser(cx.parse_sess(), tokens.iter().cloned().collect());
-    let block = parser::parse(&mut parser);
+    let block = parser::parse(&mod_name, &mut parser);
     let item = block.generate(cx, None);
 
-    let module = quote_item!(cx,
-                             #[allow(non_snake_case)]
```

Comment:
We'll still need this or the compiler shows a warning for e.g. `describe "Node" {` (this throws dozens of warnings in https://github.com/utkarshkukreti/select.rs).